rerankers/openai.py

```diff
@@ -0,0 +1,156 @@
+from __future__ import annotations
+
+import os
+from typing import Any, Dict, List, Optional, Sequence, Tuple
+from pydantic import BaseModel
+from langchain.schema import Document
+from langchain_core.callbacks import Callbacks
+from langchain.retrievers.document_compressors.base import BaseDocumentCompressor
+import logging
+import openai
+
+from settings import DEFAULT_LLM_MODEL
+
+log = logging.getLogger(__name__)
+
+
+class Ranking(BaseModel):
+    index: int
+    relevance_score: float
+    is_relevant: bool
+
+
+class OpenAIReranker(BaseDocumentCompressor):
+    _default_model: str = DEFAULT_LLM_MODEL
+
+    top_n: int = 5  # Default number of documents to return
+    model: str = DEFAULT_LLM_MODEL  # Model to use for reranking
+    temperature: float = 0.0  # Temperature for the model
+    openai_api_key: Optional[str] = None
+    remove_irrelevant: bool = True  # New flag to control removal of irrelevant documents,
```

Comment:
by default it will only include documents deemed as relevant.

If we set to false it will also include non-relevant documents